### DISCUSSION TITLE
Cross-compiling for arm-linux-gnueabihf platform

### DISCUSSION THREAD

**Discussion by jptalledo10** [protobuf-c/protobuf-c#594] [Q&A]
Hi

I am trying to cross-compile protobuf-c using the following:

 cd deps
 git clone https://github.com/google/protobuf.git protobuf
 cd protobuf
 ./autogen.sh
 ./configure --prefix=`pwd`/../`uname -m`-linux-gnu
 make
 make install
 cd ../..

 cd deps
 git clone https://github.com/protobuf-c/protobuf-c.git protobuf-c
 cd protobuf-c
 ./autogen.sh
 mkdir ../pbc-`uname -m`
 cd ../pbc-`uname -m`
 ../protobuf-c/configure --prefix=`pwd`/../`uname -m`-linux-gnu \
   PKG_CONFIG_PATH=`pwd`/../`uname -m`-linux-gnu/lib/pkgconfig
 make
 make install
 cd ../..

 cd deps
 mkdir -p pbc-arm
 cd pbc-arm
 ../protobuf-c/configure --host=arm-linux-gnueabihf --prefix=`pwd`/../arm-linux-gnueabihf \
                         --disable-protoc PATH=`pwd`/../`uname -m`-linux-gnu/bin:$PATH
 make PATH=`pwd`/../`uname -m`-linux-gnu/bin:$PATH
 make install PATH=`pwd`/../`uname -m`-linux-gnu/bin:$PATH
 cd ../..

But getting the following:

make  all-am
make[1]: Entering directory '/home/duser/mk5/stack/apps/obu-rsu/cohda/deps/pbc-cohda'
  CC       protobuf-c/protobuf-c.lo
  CCLD     protobuf-c/libprotobuf-c.la
ar: `u' modifier ignored since `D' is the default (see `U')

I tried also:

 ../protobuf-c/configure --host=arm-linux CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ --prefix=`pwd`/../arm-linux-gnueabihf --disable-protoc PATH=`pwd`/../`uname -m`-linux-gnu/bin:$PATH

But I am not getting libprotobuf.a generated for arm. I  run make check:

============================================================================
Testsuite summary for protobuf-c 1.3.3
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================

**Comment by lipnitsk** [comment]
Please search for closed issues first before opening new ones. Cross compile concerns have been answered in the past.
